### PR TITLE
Move main logic to parent Module

### PR DESCRIPTION
**Pull request by AlexWayfer**
What do you think? I'm not sure that this approach simplifies the perception, interesting your opinion.

### CODE REVIEW

File: src/update/provider.cr [hunk 1]
```diff
@@ -9,12 +9,30 @@ module Update
 
     macro extended
       PROVIDERS << {{@type}}
+
+      def executable
+        {{@type}}::EXECUTABLE
+      end
+
+      def title
+        {{@type}}::TITLE
```
Comment (keplersj) [inline src/update/provider.cr]: Maybe `SUBJECT` would be a better name for this constant?

**Comment by keplersj** [comment]
I like the way it looks. I was hoping to get a more DSL like setup eventually.

File: src/update/provider.cr (hunk 1 above)
Comment (AlexWayfer) [reply src/update/provider.cr]: Oh, what about `STUFF`? :)

But we need to write correct some banners.

For example:

``` crystal
STUFF = "Atom Packages" # => "Updating Atom Packages"
```

It looks more correct and with fewer tautology.

Comment (keplersj) [reply src/update/provider.cr]: Sounds good to me.

Comment (AlexWayfer) [reply src/update/provider.cr]: Okay, done.

**Comment by AlexWayfer** [comment]
Hey! BIG CHANGES came 😅

I hope you like it. This is what I wanted to reach.

**Comment by keplersj** [comment]
It's beautiful. Thank you

**Pull request merged**